pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -306,7 +309,25 @@ func (l *AccessCredentialManager) pingAgent(domName string) error {
 	return err
 }
 
-func (l *AccessCredentialManager) agentWriteAuthorizedKeys(domName string, user string, desiredAuthorizedKeys string) error {
+func (l *AccessCredentialManager) agentSetAuthorizedKeys(domName string, user string, authorizedKeys []string) (err error) {
+	domain, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		return err
+	}
+	defer domain.Free()
+
+	// Zero flags argument means that the authorized_keys file is overwritten with the authorizedKeys
+	err = domain.AuthorizedSSHKeysSet(user, authorizedKeys, 0)
+	if !errors.Is(err, libvirt.ERR_NO_SUPPORT) {
```

Comment:
what about use case when err == libvirt.ERR_NO_SUPPORT?
Does it has to be handled somehow?

Reply:
I missed a branch where `err == nil`. Fixed.